src/extlibs/hrvo/simulator.cpp

```diff
@@ -63,6 +62,7 @@ HRVOSimulator::HRVOSimulator(float time_step, const RobotConstants_t &robot_cons
 
 void HRVOSimulator::updateWorld(const World &world)
 {
+    this->world               = world;
```

Comment:
is this how we can set an optional variable without copying world?

Reply:
Not sure what exactly you are referring to, but this is one way to set the value of a `std::optional`. I believe that the value is copied here into the `HRVOSimulator` instance, though, based on the documentation of the [operator=](https://en.cppreference.com/w/cpp/utility/optional/operator%3D), it may also use move semantics (`std::move`) to assign the variable to `this->world`, though I'm not exactly sure if that is what's happening here.